src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+}
+```
+In this example, schema `EnterpriseUserRecord` reused `UserVisitRecord` and its annotation in the field `visitRecord`, but it overrides the annotation value of `recycledChatHistories`.
+
+
+All above examples shows that inheritance and overrides gives more extensibility to the schema annotations. Users should be free to define their own rules regarding how those annotations can be read. There should be an imambigous annotation value for the fields after user's rules are applied. We call this processing step to figure out the eventual value of a field or schema's annotation "resolution" and we call such value "resolved" value.
+
+What is more, users could have their own customized logic to process the annotations and define the custom behavior when annotations is overriden or inheritted, and even have the flexibility to set a customized resolved value by calling another local function or remote procedure.  The need for such extensibility gives the motivation for Annotation Processor. We aim to build a tool that can process annotations during schema resolution, process all interested annotations by plugin-ing user's own logic and save the "resolved" value back to corresponding DataSchema.
+
+### A common application of overriding annotation using PathSpec
+In the above `EnterpriseUserRecord` example, we use `@persistencePolicyInDays = {"/recycledChatHistories" : 3650}` do denote the field we are overriding is `recycledChatHistories`. This is basically using PathSpec to reference the field or child schema that needs to be overriden. PathSpec <insert-linke-here> can be used as a path to associate relations between fields among nested schemas. Users can use PathSpec to inambigously specifcy all the paths to the child fields that they want to override.
+
+
+In Linkedin, such usage is so common that we want to use 
+we provided `PathSpecBaseVisitor`. 
+
+It assumes \\
+(1) All *overrides* to the fields are specified using PathSpecs to the fields. \\
+(2) All *overrides* are applied on fields, not on the record
+
+For more examples regarding the syntax, one can read the java doc from `PathSpecBaseVisitor`.//<insert Link about > and the test cases //<insert Link about >
+
+
+One can seek to extend this case to adapt to their own use cases. Please see next section regarding what class to extend to fit the best use cases.
+
+## Use the schema annotation processor
+We have created a paradigm of processing Rest.Li schema annotations and wrapped them into `com.linkedin.data.schema.annotation` package.
+First thing to understand is that `DataSchema` object internally is stored recursively. We added `resolvedProperties` attribute, in order to store the final reesolved annotation.
+
+For an example schema:
+
+```pdl
+record Employee {
+  id: long
+  supervisor: Employee
+}
+```
+
+Its memory presentation is as followed:
+<center>
+<b>DataSchema Example</b><br><img src="{{ 'assets/images/DataSchema_with_resolvedProperties.png' | relative_url }}" />
```

Comment:
Should this example show different resolved values for Employee::id field?

Reply:
I think this is going to complicate this example(because different resolved value in recursive schema would be an advanced example). I really meant to show how DataSchema is stored as a reference here.